libsolidity/codegen/ContractCompiler.h

```diff
@@ -123,19 +125,35 @@ class ContractCompiler: private ASTConstVisitor
 	/// @returns the runtime assembly for clone contracts.
 	eth::AssemblyPointer cloneRuntime() const;
 
+	/// Frees the variables of a certain scope (to be used when leaving).
+	void popScopedVariables(ASTNode const* _node);
+
+	/// Pops _amount slots from the stack and jumps to _jumpTo.
+	/// Also readjusts the stack offset to the original value.
+	void popAndJump(unsigned _amount, eth::AssemblyItem const& _jumpTo);
+
+	/// Sets the stack height for the visited loop.
```

Comment:
From the name I would say that the loop is also visited. Perhaps rename to `loopVisited` or `storeLoopStackHeight`?

Reply:
Reminded me that actually that's not needed, only for loops need to